src/main/java/org/yaml/snakeyaml/constructor/SafeConstructor.java

```diff
@@ -251,7 +251,7 @@ public class ConstructYamlInt extends AbstractConstruct {
 
     @Override
     public Object construct(Node node) {
-      String value = constructScalar((ScalarNode) node).replaceAll("_", "");
```

Comment:
`replaceAll` implicitly compiles a regex on each invocation, leading to a lot of allocations and extra method invocations & CPU cycles:

![image](https://github.com/user-attachments/assets/6ead2278-ce44-44b3-8d8b-9468ef023e66)

![image](https://github.com/user-attachments/assets/f13d9b2c-4f26-4f27-9612-d81176705313)

![image](https://github.com/user-attachments/assets/db970ccb-be8c-46e9-b898-3af29918b06a)

![image](https://github.com/user-attachments/assets/b9f75be4-adce-4299-a589-7f878443eb2f)

https://private-user-images.githubusercontent.com/54594/355290326-a7e78907-bdb1-44ef-b36d-99e28d429e9c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjM3MjU4MjgsIm5iZiI6MTcyMzcyNTUyOCwicGF0aCI6Ii81NDU5NC8zNTUyOTAzMjYtYTdlNzg5MDctYmRiMS00NGVmLWIzNmQtOTllMjhkNDI5ZTljLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MTUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODE1VDEyMzg0OFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWQ4MGNjYTY3OTFkYzU5ZjlhNzhlZGE4NTZmNDhmZTk0YjBlMzQ2NWJkNzcwOTRlMjJhMjI1NzhkYzdhOWMyODcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.lPCrV5emusSHvbqtOeRU61DpDr9w01WkELLzexu2_7k